chatkit-core/src/test/kotlin/com/pusher/chatkit/cursors/CursorStoreApplyingAddedToRoomEventTest.kt

```diff
@@ -0,0 +1,160 @@
+                                roomId = "roomId1",
+                                unreadCount = 2,
+                                cursor = cursor
+                        ),
+                        membership = mock() // should not be used by CursorStore
+                ))
+            }
+
+            it("then the result indicates that the new cursor has been stored") {
+                assertThat(applyEventResult).containsExactly(UserInternalEvent.NewCursor(cursor))
+            }
+            it("then the store contains the new cursor") {
+                assertThat(subject[cursor.userId][cursor.roomId]).isEqualTo(cursor)
+            }
+        }
+        describe("when applying AddedToRoomEvent for another room with no Cursor") {
+            lateinit var applyEventResult: List<UserInternalEvent>
+            beforeEachTest {
+                applyEventResult = subject.applyEvent(UserSubscriptionEvent.AddedToRoomEvent(
+                        room = mock(), // should not be used by CursorStore
+                        readState = RoomReadStateApiType(
+                                roomId = "roomId1",
+                                unreadCount = 2,
+                                cursor = null
+                        ),
+                        membership = mock() // should not be used by CursorStore
+                ))
+            }
+
+            it("then the result indicates that nothing has been applied") {
+                assertThat(applyEventResult).isEmpty()
+            }
+            it("then the store contains the initial cursor") {
+                assertThat(subject[initialCursor.userId][initialCursor.roomId])
+                        .isEqualTo(initialCursor)
+            }
+        }
+        describe("when applying AddedToRoomEvent for another room with a Cursor") {
+            val cursor = Cursor("alice", "roomId1001", position = 2)
+
+            lateinit var applyEventResult: List<UserInternalEvent>
+            beforeEachTest {
+                applyEventResult = subject.applyEvent(UserSubscriptionEvent.AddedToRoomEvent(
+                        room = mock(), // should not be used by CursorStore
+                        readState = RoomReadStateApiType(
+                                roomId = "roomId1",
+                                unreadCount = 2,
+                                cursor = cursor
+                        ),
+                        membership = mock() // should not be used by CursorStore
+                ))
+            }
+
+            it("then the result indicates that the new cursor has been stored") {
+                assertThat(applyEventResult).containsExactly(UserInternalEvent.NewCursor(cursor))
+            }
+            it("then the store contains the new cursor") {
+                assertThat(subject[cursor.userId][cursor.roomId]).isEqualTo(cursor)
+            }
+            it("then the store contains the initial cursor") {
```

Comment:
would it be clearer to say "the store still contains the cursor for the other room", or "the cursors for the other room was unaffected in the store"?

Reply:
I think it's clearer and moreover more concise in it's current form as when it comes to the suggestion it would not be clear which room is "the other". Also the statements "still contains" or "was unaffected" do not correspond to what this test is checking as this test is not checking the store's initial state (that would not be correct in BDD to do assertions before the when part). Indeed those assertions for initial state are effectively made in the first group of tests where the given is empty store (or could be made by testing of initialiseContents fun but it's existing and trivial logic).